python/modem-firmware-1.3+/device_credentials_installer.py

```diff
@@ -605,96 +625,87 @@ def main():
             write_line('at enable')
             wait_for_prompt(b'to exit AT mode')
 
-    # prepare modem so we can interact with security keys
-    print(local_style('Disabling LTE and GNSS...'))
-    write_line('AT+CFUN=4')
+    cred_if = None
+    if args.cmd_type == CMD_TYPE_AT:
+        cred_if = ATCommandInterface(write_line, wait_for_prompt, verbose)
 
-    retval = wait_for_prompt(b'OK')
-    if not retval:
-        print(error_style('Unable to communicate'))
-        cleanup()
-        sys.exit(6)
+    if args.cmd_type == CMD_TYPE_TLS_SHELL:
+        cred_if = TLSCredShellInterface(write_line, wait_for_prompt, verbose)
+
+    # prepare modem so we can interact with security keys
+    if (cmd_type_has_at):
+        print(local_style('Disabling LTE and GNSS...'))
+        if not cred_if.go_offline():
+            print(error_style('Unable to communicate'))
+            cleanup()
+            sys.exit(6)
 
     # get the IMEI
-    write_line('AT+CGSN')
-    retval, output = wait_for_prompt(b'OK', b'ERROR', store=b'\r\n')
-    if not retval:
+    imei = ""
+    if (cmd_type_has_at):
+        imei = cred_if.get_imei()
+
+    if imei is None:
         print(error_style('Failed to obtain IMEI'))
         cleanup()
         sys.exit(7)
+
     # display the IMEI for reference
-    imei = str(output.decode("utf-8"))[:IMEI_LEN]
     print(hivis_style('Device IMEI: ' + imei))
 
     # get and verify the modem firmware version
-    mfw_ver = check_mfw_version()
+    mfw_ver = ""
+    if (cmd_type_has_at):
+        mfw_ver = check_mfw_version()
 
     # set custom device ID
-    custom_dev_id = args.id_str
+    desired_dev_id = args.id_str
```

Comment:
I renamed `custom_dev_id` because I found it confusing when sharing scope with `dev_id` -- `desired` is supposed to outline that `desired_dev_id` is essentially the ID we request, whereas `dev_id` is the one we ultimately get

As for the inconsistency, I can rename the other usages of `custom` to `desired`. Unless you would prefer that I go back to using `custom_dev_id` instead of `desired_dev_id`

Reply:
i would just leave it `custom_dev_id`.  it's only appears 3 times and that way other changes aren't needed.